backend/routes/instructors.js

```diff
@@ -1,2119 +1,122 @@
+
+router.get('/', function (req, res) {
+	let instructorsAll = instructorsData.getInstructors();
+
+	res.status(200).send(instructorsAll);
 });
 
-// READ
-router.get('instructor/:id', function (req, res, next) {
-	const foundInstructor = instructors.find(
-		(user) => user.id.$oid === req.params.id
-	);
+router.get('/:id', function (req, res) {
 
-	if (!foundInstructor)
-		return res.status(404).send({ message: 'User not found' });
+	const id = req.params.id;
+	const instructor = instructorsData.getInstructorById(id);
 
-	return res.send(foundInstructor);
+	(instructor)
+		? res.status(200).send(instructor)
+		: res.status(404).send({ message: `User ${id} not found` })
 });
 
-// CREATE
-router.post('/', function (req, res, next) {
-	if (!req.body.first_name || !req.body.last_name) {
-		return res
-			.status(404)
-			.send({ message: 'Name can not be empty' });
-	}
-	if (!req.body.password) {
-		return res
-			.status(404)
-			.send({ message: 'Password can not be empty' });
-	}
-	if (!req.body.email) {
-		return res
-			.status(404)
-			.send({ message: 'Email can not be empty' });
-	}
-	if (!req.body.phone) {
-		return res
-			.status(404)
-			.send({ message: 'Phone can not be empty' });
-	}
-	if (!req.body.street || !req.body.city || !req.body.country) {
-		return res
-			.status(404)
-			.send({ message: 'Address can not be empty' });
-	}
-	if (!req.body.company) {
-		return res
-			.status(404)
-			.send({ message: 'Company can not be empty' });
-	}
-	if (!req.body.language) {
-		return res
-			.status(404)
-			.send({ message: 'Language can not be empty' });
-	}
-	if (!req.body.experience) {
-		return res
-			.status(404)
-			.send({ message: 'Experience can not be empty' });
-	}
-	if (!req.body.license) {
-		return res
-			.status(404)
-			.send({ message: 'License can not be empty' });
-	}
-	if (!req.body.time) {
-		return res
-			.status(404)
-			.send({ message: 'Time can not be empty' });
-	}
-
-	const instructor = {
-		id: { $oid: uuid() },
+router.post('/', function (req, res) {
+	const inputInstructor = {
 		first_name: req.body.first_name,
 		last_name: req.body.last_name,
 		password: req.body.password,
 		email: req.body.email,
 		phone: req.body.phone,
-		gender: req.body.gender ? req.body.gender : 'None',
-		photo: req.body.photo
-			? req.body.photo
-			: 'https://picsum.photos/200',
-		Rating: 0,
 		street: req.body.street,
 		city: req.body.city,
 		country: req.body.country,
 		company: req.body.company,
-		language: req.body.language,
+		language: req.body.language ? req.body.language : 'English',
 		experience: req.body.experience,
 		license: req.body.license,
-		description: req.body.description
-			? req.body.description
-			: 'No description for this instructor',
 		time: req.body.time,
-		carIsProvided: req.body.carIsProvided
-			? req.body.carIsProvided
-			: 'false',
-		reviews: [],
-	};
-	instructors.push(instructor);
-	return res.send(instructor);
+	}
+
+	try {
+		validateInstructor(inputInstructor);
```

Comment:
As mentioned, do this validation in the service function

Reply:
Basically wrap the `instructorsService.addInstructor` call in a try catch. In the catch, that means validate failed, then you can return your 400. If success, but undefined, then you can return your 424